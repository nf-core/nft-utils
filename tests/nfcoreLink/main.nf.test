nextflow_process {

    name "Test Process TEST_MODULE"
    script "./main.nf"
    process "TEST_MODULE"

    tag "test"

    test("test plugin") {

        setup {
            nfcoreInitialise("${launchDir}/library/")
            nfcoreInstall("${launchDir}/library/", ["minimap2/index"])
            nfcoreLink("${launchDir}/library/", baseDir)
        }

        then {
            def minimap2_index_files = getAllFilesFromDir("${baseDir}/nf-core/minimap2/index", relative: true)
            assertAll(
                { assert snapshot(minimap2_index_files).match() }
            )
        }

        cleanup {
            nfcoreUnlink("${launchDir}/library/", baseDir)
        }

    }

}
