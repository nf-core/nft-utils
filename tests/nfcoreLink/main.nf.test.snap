{
    "test plugin": {
        "content": [
            [
                "environment.yml",
                "main.nf",
                "meta.yml",
                "tests/main.nf.test",
                "tests/main.nf.test.snap"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-26T15:34:09.729923"
    }
}